### PR TITLE
Remove ::align() from OMR compiler

### PR DESCRIPTION
**Pull request by ymanton**
This patch removes ::align() altogether from the OMR compiler.

This is a follow-up to https://github.com/eclipse/omr/pull/4960.

### CODE REVIEW

**Comment by 0xdaryl** [comment]
AppVeyor looks like a real build failure to investigate.  I'll launch a genie build so you see how the other platforms fare as well.

**Comment by 0xdaryl** [comment]
@genie-omr build all

**Comment by ymanton** [comment]
Thanks. Some of the failures are caused by leftover references to `Alignment.cpp` in some makefiles, but some are caused by the code in `Bytes.hpp` being incompatible with MSVC. I'll look into that.

**Comment by ymanton** [comment]
MSVC issue turned out to be a more general Windows issue. Fixed it via https://github.com/eclipse/omr/pull/4977/commits/1c4df21f4d88f0ee94b20c76d2a1d7a1c16cc90a, rationale is in the commit message.

**Comment by ymanton** [comment]
@0xdaryl can you run PR testing on this?

**Comment by 0xdaryl** [comment]
@genie-omr build all

**Comment by ymanton** [comment]
ZOS failed to launch.

```
10:47:23  Fetching upstream changes from git://github.com/eclipse/omr.git
10:47:23   > /u/jenkins/jenkins-zos/git --version # timeout=10
10:47:23   > /u/jenkins/jenkins-zos/git fetch --tags --progress git://github.com/eclipse/omr.git +refs/pull/*:refs/remotes/origin/pr/* # timeout=10
10:53:35  ERROR: Error fetching remote repo 'origin'
10:53:35  hudson.plugins.git.GitException: Failed to fetch from git://github.com/eclipse/omr.git
```

**Comment by 0xdaryl** [comment]
@genie-omr build zos

File: compiler/runtime/CMakeLists.txt [hunk 1]
```diff
@@ -22,7 +22,6 @@
 compiler_library(runtime
 	${CMAKE_CURRENT_LIST_DIR}/Runtime.cpp
 	${CMAKE_CURRENT_LIST_DIR}/Trampoline.cpp
-	${CMAKE_CURRENT_LIST_DIR}/Alignment.cpp
```
Comment (0xdaryl) [inline compiler/runtime/CMakeLists.txt:25]: Please update copyright header.

File: jitbuilder/build/files/common.mk [hunk 2]
```diff
@@ -238,7 +238,6 @@ JIT_PRODUCT_BACKEND_SOURCES+=\
     $(JIT_OMR_DIRTY_DIR)/ilgen/OMRVirtualMachineRegister.cpp \
     $(JIT_OMR_DIRTY_DIR)/ilgen/OMRVirtualMachineRegisterInStruct.cpp \
     $(JIT_OMR_DIRTY_DIR)/ilgen/OMRVirtualMachineState.cpp \
-    $(JIT_OMR_DIRTY_DIR)/runtime/Alignment.cpp \
```
Comment (0xdaryl) [inline jitbuilder/build/files/common.mk:241]: Please update copyright header.

**Comment by ymanton** [comment]
Copyrights updated.

**Comment by fjeremic** [comment]
@genie-omr build all

**Pull request merged**